aws/src/test/java/org/apache/iceberg/aws/s3/TestFuzzyS3InputStream.java

```diff
@@ -0,0 +1,305 @@
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
+        },
+        exception,
+        shouldRetry);
+  }
+
+  @ParameterizedTest
+  @MethodSource("parameters")
+  public void testSeekWithFuzzyStreamRetrySucceed(Exception exception, boolean shouldRetry)
+      throws Exception {
+    Assumptions.assumeTrue(shouldRetry);
+    S3FileIOProperties awsProperties = new S3FileIOProperties();
+    awsProperties.setS3RetryNumRetries(4);
+    awsProperties.setS3RetryMinWaitMs(100);
+    testSeek(fuzzyStreamClient(new AtomicInteger(3), (IOException) exception), awsProperties);
+  }
+
+  @ParameterizedTest
+  @MethodSource("parameters")
+  public void testSeekWithFuzzyStreamRetryFail(Exception exception, boolean shouldRetry)
+      throws Exception {
+    Assumptions.assumeTrue(shouldRetry);
+    testRetryFail(
+        (counter, awsProperties) -> {
+          try {
+            testSeek(fuzzyStreamClient(counter, (IOException) exception), awsProperties);
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
+        },
+        exception,
+        shouldRetry);
+  }
+
+  @ParameterizedTest
+  @MethodSource("parameters")
+  public void testRangeReadWithFuzzyStreamRetrySucceed(Exception exception, boolean shouldRetry)
+      throws Exception {
+    Assumptions.assumeTrue(shouldRetry);
+    S3FileIOProperties awsProperties = new S3FileIOProperties();
+    awsProperties.setS3RetryNumRetries(4);
+    awsProperties.setS3RetryMinWaitMs(100);
+    testRangeRead(fuzzyStreamClient(new AtomicInteger(3), (IOException) exception), awsProperties);
+  }
+
+  @ParameterizedTest
+  @MethodSource("parameters")
+  public void testRangeReadWithFuzzyStreamRetryFail(Exception exception, boolean shouldRetry)
+      throws Exception {
+    Assumptions.assumeTrue(shouldRetry);
+    testRetryFail(
+        (counter, awsProperties) -> {
+          try {
+            testRangeRead(fuzzyStreamClient(counter, (IOException) exception), awsProperties);
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
+        },
+        exception,
+        shouldRetry);
+  }
+
+  private void testRetryFail(
+      BiConsumer<AtomicInteger, S3FileIOProperties> runnable,
+      Exception exception,
+      boolean shouldRetry)
+      throws Exception {
+    Assumptions.assumeTrue(shouldRetry);
+    S3FileIOProperties awsProperties = new S3FileIOProperties();
+    awsProperties.setS3RetryNumRetries(2);
+    awsProperties.setS3RetryMinWaitMs(100);
+    AtomicInteger counter = new AtomicInteger(4);
+
+    org.assertj.core.api.Assertions.assertThatThrownBy(
+            () -> runnable.accept(counter, awsProperties))
+        .as("Should fail after retries")
+        .cause()
+        .isInstanceOf(exception.getClass())
+        .hasMessageContaining(exception.getMessage());
+    Assertions.assertEquals(
```

Comment:
please use AssertJ assertions everywhere

Reply:
Will exclude all InputStream changes from this PR in favor of https://github.com/apache/iceberg/pull/10433. Will double check on that PR to make sure we use assertJ everywhere